### ISSUE TITLE
SystemObserver.getAppVersion crash

### ISSUE THREAD

**Issue by matteinn** [BranchMetrics/android-branch-deep-linking#680]
Hi all!
I'm getting a rare crash on the latest version (3.0.4) of the Android SDK, older versions are affected as well. Is there a way I can prevent this?
Thank you!

```
Fatal Exception: java.lang.RuntimeException: Package manager has died
android.app.ApplicationPackageManager.getPackageInfo (ApplicationPackageManager.java:137)
--
  | io.branch.referral.SystemObserver.getAppVersion (SystemObserver.java:131)
  | io.branch.referral.ServerRequestInitSession.setPost (ServerRequestInitSession.java:49)
  | io.branch.referral.ServerRequestRegisterOpen. (ServerRequestRegisterOpen.java:34)
  | io.branch.referral.Branch.getInstallOrOpenRequest (Branch.java:2413)
  | io.branch.referral.Branch.registerAppInit (Branch.java:2372)
  | io.branch.referral.Branch.initializeSession (Branch.java:2340)
  | io.branch.referral.Branch.initUserSessionInternal (Branch.java:1308)
  | io.branch.referral.Branch.initSession (Branch.java:952)
  | io.branch.referral.Branch.initSessionWithData (Branch.java:1090)
  | io.branch.referral.Branch.startSession (Branch.java:2638)
  | io.branch.referral.Branch.access$100 (Branch.java:72)
  | io.branch.referral.Branch$BranchActivityLifeCycleObserver.safedk_c$a_onActivityStarted_874d29069972b2985395a53bc37efaec (Branch.java:2571)
  | io.branch.referral.Branch$BranchActivityLifeCycleObserver.onActivityStarted (Branch.java)
  | android.app.Application.dispatchActivityStarted (Application.java:205)
  | android.app.Activity.onStart (Activity.java:1155)
  | android.support.v4.app.FragmentActivity.onStart (FragmentActivity.java:614)
  | android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1240)
  | android.app.Activity.performStart (Activity.java:6322)
  | android.app.Activity.performRestart (Activity.java:6368)
  | android.app.ActivityThread.handleSleeping (ActivityThread.java:3660)
  | android.app.ActivityThread.-wrap19 (ActivityThread.java)
  | android.app.ActivityThread$H.handleMessage (ActivityThread.java:1520)
  | android.os.Handler.dispatchMessage (Handler.java:102)
  | android.os.Looper.loop (Looper.java:148)
  | android.app.ActivityThread.main (ActivityThread.java:5452)
  | java.lang.reflect.Method.invoke (Method.java)
  | com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:781)
  | com.android.internal.os.ZygoteInit.main (ZygoteInit.java:671)
```

**Comment by code-crusher** [comment]
Hey @matteinn thanks for the logs. We are looking into this and would get back to you.

**Comment by csalmi-branch** [comment]
@matteinn We released a fix for this issue as part of v3.1.2.

**Comment by matteinn** [comment]
Thank you so much!
